engine/util.go

```diff
@@ -58,7 +58,7 @@ func writeEnv(w io.Writer, os, key, value string) {
 		fmt.Fprintf(w, "$Env:%s = %q", key, value)
 		fmt.Fprintln(w)
 	default:
-		fmt.Fprintf(w, "export %s=%q", key, value)
+		fmt.Fprintf(w, "export %s='%s'", key, value)
```

Comment:
I am not sure this is the best fix, but figured this would be a good start to get the conversation going.

Reply:
the reason we use `'%q'` instead of `'%s'` is because it escapes most special characters. So even though this change could improve handling backticks, it would cause a regression when a string containers other characters like newlines, tabs, etc.

Reply:
Yeah, I figured double quotes were used for good reason. The only alternative I can think of is to parse an env var, and escape back ticks, if not already. That feels gross. I don't think users should have to avoid using backticks in their PR templates. It is common to have things `formatted`.

I can think of an acceptable alternative. We can just use "```". Still leaves an edge case here there we never experienced with other runners.

I am open to alternative fixes here. Otherwise, I can try the triple backticks.